src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2559,6 +2645,23 @@ private static object FoldCheckedIntegralUnaryOperator(UnaryOperatorKind kind, C
             return null;
         }
 
+        private static object FoldNativeIntegerUnaryOperator(UnaryOperatorKind kind, ConstantValue value)
+        {
+            checked
+            {
+                switch (kind)
+                {
+                    case UnaryOperatorKind.NIntUnaryMinus:
+                        return -value.Int32Value;
+                    case UnaryOperatorKind.NIntBitwiseComplement:
+                    case UnaryOperatorKind.NUIntBitwiseComplement:
+                        throw new OverflowException();
```

Comment:
>throw new OverflowException(); [](start = 24, length = 30)

This is unexpected. These operators do not overflow, we simply unable to fold, therefore should return null.